src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -1039,3 +1038,95 @@ const Json::Value& CascadiaSettings::_GetDisabledProfileSourcesJsonObject(const
     }
     return json[JsonKey(DisabledProfileSourcesKey)];
 }
+
+// Method Description:
+// - Create a timestamped backup file with the current contents
+// Arguments:
+// - content - the content that we're writing to the backup file
+// - settingsPath - the path to the settings file that we're going to create a backup for
+// Return Value:
+// - <none>
+void CascadiaSettings::_WriteBackupFile(std::string_view content, const winrt::hstring settingsPath)
+{
+    // create a timestamped backup file
+    const auto clock{ std::chrono::system_clock() };
+    const auto timeStamp{ clock.to_time_t(clock.now()) };
+    const auto backupSettingsPath{ fmt::format(L"{}.{:%Y-%m-%dT%H:%M:%S}.backup", settingsPath, fmt::localtime(timeStamp)) };
+
+    wil::unique_hfile backupFile{ CreateFileW(backupSettingsPath.c_str(),
+                                              GENERIC_READ,
+                                              FILE_SHARE_READ | FILE_SHARE_WRITE,
+                                              nullptr,
+                                              CREATE_NEW,
+                                              FILE_ATTRIBUTE_NORMAL,
+                                              nullptr) };
+
+    // throw if backup file already exists
+    THROW_LAST_ERROR_IF(INVALID_HANDLE_VALUE == backupFile.get());
+    THROW_LAST_ERROR_IF(!WriteFile(backupFile.get(), content.data(), gsl::narrow<DWORD>(content.size()), nullptr, nullptr));
+}
+
+// Method Description:
+// - Write the current state of CascadiaSettings to our settings file
+// - Create a backup file with the current contents, if one does not exist
+// Arguments:
+// - <none>
+// Return Value:
+// - <none>
+void CascadiaSettings::WriteSettingsToDisk() const
+{
+    const auto settingsPath{ CascadiaSettings::SettingsPath() };
+
+    try
+    {
+        _WriteBackupFile(_userSettingsString, settingsPath);
+    }
+    CATCH_LOG();
+
+    // write current settings to current settings file
+    const auto json{ ToJson() };
+    _WriteSettings(json.toStyledString(), settingsPath);
+}
+
+// Method Description:
+// - Create a new serialized JsonObject from an instance of this class
+// Arguments:
+// - <none>
+// Return Value:
+// the JsonObject representing this instance
+Json::Value CascadiaSettings::ToJson() const
+{
+    // top-level json object
+    // directly inject "globals" and "$schema" into here
+    Json::Value json{ _globals->ToJson() };
+    JsonUtils::SetValueForKey(json, SchemaKey, JsonKey(SchemaValue));
+
+    // "profiles" will always be serialized as an object
+    Json::Value profiles{ Json::ValueType::objectValue };
+    profiles[JsonKey(DefaultSettingsKey)] = _userDefaultProfileSettings->ToJson();

```

Comment:
umm..

if the user doesn't have a defaults block this is 100% going to crash my dude

Reply:
i'm going to need you to test both this and the legacy profiles layout (array only)

Reply:
Latest commit verifies that this doesn't crash :) Good catch.